src/update/UpdateManager.cpp

```diff
@@ -62,48 +60,50 @@ VersionPatch UpdateManager::getVersionPatch() const
 
 void UpdateManager::getUpdates()
 {
-    audacity::network_manager::Request request("https://updates.audacityteam.org/feed/latest.xml");
+    const audacity::network_manager::Request request("https://updates.audacityteam.org/feed/latest.xml");
     auto response = audacity::network_manager::NetworkManager::GetInstance().doGet(request);
 
     response->setRequestFinishedCallback([response, this](audacity::network_manager::IResponse*) {
 
+        wxFrame* parent = FindProjectFrame(GetActiveProject());
+        wxASSERT(parent);
+
+        if (!parent) return;
```

Comment:
Do not assert (parent).  It may be null, on Mac when there is no open window, and that is not because of any logical error in the program or failure to observe pre- and post-conditions.

As we discussed elsewhere, it is safe to make modal dialogs like the exception dialogs below with no parent.  So you do not need the early return.